packages/config/src/tokens/types.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  AssetId,
+  branded,
+  CoingeckoId,
+  EthereumAddress,
+  UnixTime,
+} from '@l2beat/shared'
+import { z } from 'zod'
+
```

Comment:
On Starkex we have helper functions for branded types
```tsx
export function stringAs<T>(Brand: (s: string) => T) {
  return z
    .string()
    .refine((s) => {
      try {
        Brand(s)
        return true
      } catch {
        return false
      }
    })
    .transform(Brand)
}
```

Then you would use it like this
```tsx
id: stringAs(AssetId)
```

Maybe you can add it on L2Beat also

Reply:
You can find those helper functions there: starkex-explorer/packages/shared/src/types.ts